src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -431,6 +431,18 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
         }
     }
 
+    for (std::size_t i = 0u; i < orderedSubgraphs.size(); i++) {
+        if (!m_compiled_submodels[i].replaced_by) {
+            continue;
+        } else {
+            auto real_id = m_compiled_submodels[i].replaced_by.value();
+            if (i != real_id) {
+                auto comp_desc = m_compiled_submodels[real_id];
+                m_compiled_submodels[i].device_it = m_compiled_submodels[real_id].device_it;
+            }
+        }
+    }
```

Comment:
Please don't do this too (see above).

The access to device iterator must be correct in all places. If it is not correct somewhere, _that_ place should be fixed.

Reply:
Fixed, thanks!